modular/gater/request_context.go

```diff
@@ -266,23 +293,51 @@ func (r *RequestContext) verifyOffChainSignature(requestSignature string) (sdk.A
 	}
 }
 
+// verifyOffChainSignatureFromPreSignedURL used to verify off-chain-auth signature, return (address, nil) if check succeed.  The auth information will be parsed from URL.
+func (r *RequestContext) verifyGNFD1EddsaSignatureFromPreSignedURL(authenticationStr string, account string, domain string) (sdk.AccAddress, error) {
+	var (
+		err error
+	)
+	_, sigString, err := parseSignedMsgAndSigFromRequest(authenticationStr)
+	if err != nil {
+		return nil, err
+	}
+
+	// check request integrity
+	realMsgToSign := commonhttp.GetMsgToSignInGNFD1AuthForPreSignedURL(r.request)
+
+	offChainSig := *sigString
+
+	verifyOffChainSignatureResp, err := r.g.baseApp.GfSpClient().VerifyGNFD1EddsaSignature(r.Context(), account, domain, offChainSig, realMsgToSign)
+	if err != nil {
+		log.Errorf("failed to verify off chain signature", "error", err)
+		return nil, err
+	}
+	if verifyOffChainSignatureResp {
+		userAddress, _ := sdk.AccAddressFromHexUnsafe(account)
+		return userAddress, nil
+	} else {
+		return nil, err
+	}
+}
+
 func parseSignedMsgAndSigFromRequest(requestSignature string) (*string, *string, error) {
 	var (
 		signedMsg string
 		signature string
 	)
 	requestSignature = strings.ReplaceAll(requestSignature, "\\n", "\n")
 	signatureItems := strings.Split(requestSignature, ",")
-	if len(signatureItems) != 2 {
-		return nil, nil, ErrAuthorizationHeaderFormat
-	}
+
 	for _, item := range signatureItems {
 		pair := strings.Split(item, "=")
 		if len(pair) != 2 {
 			return nil, nil, ErrAuthorizationHeaderFormat
 		}
 		switch pair[0] {
 		case SignedMsg:
+			// for off-chain-auth, client doesn't need to pass the signedMsg, as SP will generate the signed msg by itself.
+			// but this method is also used by updateUserPublicKeyHandler API, which uses the personal sign verification and require clients to send a signed Msg.
```

Comment:
Why do we have to put two fields in one authorization header, is there any best practice to follow, or it is our own invent?